sklearn/ensemble/gradient_boosting_cv.py

```diff
@@ -0,0 +1,804 @@
+    of parameters. In most cases, this will be much faster than using
+    :class:`sklearn.model_selection.GridSearchCV` due to early stopping.
+    All parameters which are not parsed by this class are passed directly to
+    the underlying Gradient Boosting estimator. If one or more parameters is a
+    list, the Gradient Boosting model is evaluated for all elements of the
+    list.
+
+    Parameters
+    ----------
+    n_stop_rounds : int, optional, default=10
+        If the score on the test set rounded off to `score_precision` decimal
+        places does not change for `n_stop_rounds` iterations, the gradient
+        boosting is halted.
+
+    score_precision : int, optional, default=2
+        The number of decimal places to round the score off by before
+        comparing it with the scores of previous iterations.
+
+    max_iter : int, optinal, default=10000
+        The maximum number of estimators that will be added to the Gradient
+        Boosting class. If the model converges, the number of estimators will
+        be lower than this.
+
+    cv : int, cross-validation generator or an iterable, optional
+        Determines the cross-validation splitting strategy.
+        Possible inputs for cv are:
+          - None, to use the default 3-fold cross validation,
+          - integer, to specify the number of folds in a `(Stratified)KFold`,
+          - An object to be used as a cross-validation generator.
+          - An iterable yielding train, validation splits.
+
+        For integer/None inputs, ``StratifiedKFold`` is used for classification
+        tasks, when ``y`` is binary or multiclass.
+
+        See the :mod:`sklearn.model_selection` module for the list of
+        cross-validation strategies that can be used here.
+
+        Also refer :ref:`cross-validation documentation <cross_validation>`
+
+    scoring : string, callable or None, default=None
+        A string (see model evaluation documentation) or
+        a scorer callable object / function with signature
+        ``scorer(estimator, X, y)``.
+        If ``None``, the ``score`` method of the estimator is used.
+
+    refit : boolean, default=True
+        If set, the data will be fit on a new Gradient Boosting model with
+        the parameters that gave the best mean score. If not set, it is not
+        possbile to make predections later with this class.
+
+    n_jobs : int, default=1
+        Number of jobs to run in parallel.
+
+    verbose : integer
+        Controls the verbosity: the higher, the more messages.
+
+    loss : string, list of strings, optional (default='deviance')
+        loss function to be optimized. 'deviance' refers to
+        deviance (= logistic regression) for classification
+        with probabilistic outputs. For loss 'exponential' gradient
+        boosting recovers the AdaBoost algorithm. If a list is given, all
+        loss functions are evaluated to choose the best one.
```

Comment:
`loss` is considered as any other hyper-parameter. This is the loss of the underlying GB model.